### ISSUE TITLE
Important tests being skipped

### ISSUE THREAD

**Issue by matt-snider** [mongomock/mongomock#237]
Hi! 

I'm new to the project, and was looking to make some contributions to aggregation pipeline support that I think would be quite useful. 

As I've been looking through the code, I noticed that some tests are being skipped by tox in both the pymongo and pyexecjs environments. This is due to some tests have both `skipIf()` decorators, for example:

``` python
@skipIf(not _HAVE_PYMONGO, "pymongo not installed")
@skipIf(not _HAVE_MAP_REDUCE, "execjs not installed")
class MongoClientAggregateTest(_CollectionComparisonTest):
      #....
```

It seems to me that the simplest solution would just be to modify the tox.ini deps to include pyexecjs in the pymongo environment, and add verbose logging to the nosetest call to make it easier to recognize when something is being incorrectly skipped.

I wanted to get in touch before opening a PR in case there's a reason it's done this way. :smile:

**Comment by texuf** [comment]
Hey sorry nobody ever got back to you. I'm sure the tests were skipped before we even used Tox.

Were you able to get this done?

**Comment by matt-snider** [comment]
Hi @texuf! No worries. 

I did some work on it but didn't get around to finishing it. I'll take a look now and possibly open a PR

**Comment by matt-snider** [comment]
@texuf Finished the PR last week, but there are some issues with tests from the last few PRs failing since these tests were never actually run. They should pass if tox is run with a newer Mongo version. See #265

**Comment by smalekzadeh** [comment]
Hi 
Can anyone help please? Does mongomock has support for test aggregate pipelines?
Thanks

**Comment by smalekzadeh** [comment]
I get the "TypeError: 'Collection' object is not callable" error when try to use it

**Comment by pcorpet** [comment]
Thanks for reporting and sorry for the long delay in the response. This was eventually fixed by ac610fc and then further down by bd99411.
